packages/compass-connections/src/stores/connections-store-redux.ts

```diff
@@ -1556,7 +1559,9 @@ const connectWithOptions = (
               ),
               connectionInfo: {
                 id: connectionInfo.id,
-                isAtlas: !!connectionInfo.atlasMetadata,
+                isAtlas: isAtlas(
+                  connectionInfo.connectionOptions.connectionString
+                ),
```

Comment:
Can this be completely encapsulated inside the `adjustConnectionOptionsBeforeConnect` method? Is there any reason to keep this logic configurable from outside if we always want this to be applied based on the connection string?